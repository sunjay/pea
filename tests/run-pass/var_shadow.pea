fn main() {
    let a = 3;
    println!(a);
    // Shadow variable `a`
    let a = 4;
    println!(a);
    // This is fine because it's using the previously defined `a`
    let a = a;
    println!(a);

    // Functions are first class and can be printed like anything else
    println!(foo);
    println!(foo());

    // Function names can be shadowed by variables
    let foo = b"cool";
    println!(foo);
    let main = 3249;
    println!(main);
}

fn foo() {
    // Shadows the `main` function
    let main = b"wow!";
    println!(main);
}
