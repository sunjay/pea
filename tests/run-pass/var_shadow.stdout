3
4
4
<fn foo>
wow!
()
cool
3249
